python_modules/dagster/dagster/_core/definitions/asset_reconciliation_sensor.py

```diff
@@ -0,0 +1,332 @@
+
+                if materialization_planned_event_records:
+                    in_progress = context.instance.get_runs(
+                        filters=RunsFilter(
+                            run_ids=[
+                                materialization_planned_event_records[0].event_log_entry.run_id
+                            ],
+                            statuses=IN_PROGRESS_RUN_STATUSES,
+                        )
+                    )
+                    if in_progress:
+                        # we don't want to materialize the child asset because one of its parents is
+                        # being materialized. We'll materialize the asset on the next tick when the
+                        # parent is up to date
+                        parent_asset_event_records = {pp: (False, 0.0) for pp in parent_assets}
+
+                        return parent_asset_event_records
+
+            # the parent is not currently being materialized, so check if there is has a completed materialization
+            event_records = context.instance.get_event_records(
+                EventRecordsFilter(
+                    event_type=DagsterEventType.ASSET_MATERIALIZATION,
+                    asset_key=p,
+                ),
+                ascending=False,
+                limit=1,
+            )
+            print(f"materialization record {event_records}")
+
+            if event_records and event_records[0].event_log_entry.timestamp > cursor_timestamp:
+                # if the run for this ^ materialization also materialized the downstream asset, we don't consider
+                # the parent asset "updated" (but we do want to update the cursor)
+                other_materialized_assets = context.instance.get_records_for_run(
+                    run_id=event_records[0].event_log_entry.run_id,
+                    of_type=DagsterEventType.ASSET_MATERIALIZATION_PLANNED,
+                )
+                other_materialized_assets = [
+                    event.event_log_entry.dagster_event.event_specific_data.asset_key
+                    for event in other_materialized_assets.records
+                ]
+                if current_asset in other_materialized_assets:
+                    parent_asset_event_records[p] = (
+                        False,
+                        event_records[0].event_log_entry.timestamp,
+                    )
+                else:
+                    # current asset was not updated along with the parent asset, so we consider the
+                    # parent asset updated
+                    parent_asset_event_records[p] = (
+                        True,
+                        event_records[0].event_log_entry.timestamp,
+                    )
+            else:
+                # the parent has not been materialized and will not be materialized by the sensor
+                parent_asset_event_records[p] = (False, 0.0)
+
+    return parent_asset_event_records
+
+
+def _make_sensor(
```

Comment:
As this isn't user-facing, and the only place it's called is with all its values specified, better not to give it default values. Synchronizing them with the default values in other methods creates room for error.